### PR TITLE
Services: NetworkServer service

### PR DESCRIPTION
**Pull request by sppmacd**
This service is responsible for loading network configuration from a /etc/Network.ini config file. It sets up static IP address + mask or starts DHCPClient depending on configuration.

The DHCPClient is now run only by NetworkServer and only when there are interfaces that have DHCP enabled. It gets a list of interfaces that it should do DHCP discovery on.

**[DHCPClient: Don't discover interfaces other than given by default](https://github.com/SerenityOS/serenity/commit/2c1e8afcfc6388ccbf6ced66f77127c47f0b8e4e)**

Now, the caller needs to give interface names in command-line arguments.
The DHCPClient will perform DHCP discovery only on these adapters. The
service now immediately closes when no interfaces were given.

We don't check if interface has already IP address assigned; we just
reset it to zero so that DHCP resolution will not fail.

**[NetworkServer: Add a new NetworkServer service](https://github.com/SerenityOS/serenity/commit/89fd68ba42337ecfd861001cc61602c96ea87b1e)**

This service is responsible for loading network configuration from a
/etc/Network.ini config file. It sets up static IP address + mask or
starts DHCPClient depending on configuration.

**[Base: Add man page for network configuration](https://github.com/SerenityOS/serenity/commit/7049ff0620ad868cf2fdf578bdd596d6b1c5a325)**

### CODE REVIEW

File: Userland/Services/DHCPClient/main.cpp [hunk 1]
```diff
@@ -9,15 +9,22 @@
...
     TRY(Core::System::pledge("stdio unix inet cpath rpath"));
     Core::EventLoop event_loop;
 
     TRY(Core::System::unveil("/proc/net/", "r"));
     TRY(Core::System::unveil(nullptr, nullptr));
 
-    auto client = TRY(DHCPv4Client::try_create());
+    Vector<String> interfaces;
+    for (size_t i = 1; i < args.strings.size(); i++)
+        interfaces.append(args.strings[i]);
```
Comment (linusg) [inline Userland/Services/DHCPClient/main.cpp:26]: Can we use `ArgsParser` for this please?

File: Base/etc/SystemServer.ini [hunk 2]
```diff
@@ -51,6 +51,10 @@ SystemModes=text,graphical
 MultiInstance=true
 AcceptSocketConnections=true
 
+[NetworkServer]
+User=root
+SystemModes=text,graphical,self-test,generate-manpages
```
Comment (linusg) [inline Base/etc/SystemServer.ini:56]: Why would `generate-manpages` need networking? :thinking:

**Review by linusg — COMMENTED** [review]
Nice work!

File: Base/etc/SystemServer.ini (hunk 2 above)
Comment (sppmacd) [reply Base/etc/SystemServer.ini]: I just copied that from LookupServer previously, fixed! :)

File: Userland/Services/DHCPClient/main.cpp (hunk 1 above)
Comment (sppmacd) [reply Userland/Services/DHCPClient/main.cpp]: Fixed.

**Pull request merged**